### ISSUE TITLE
TimeOrdered UUID creator generate out-of-order values

### ISSUE THREAD

**Issue by tri-bao** [f4b6a3/uuid-creator#69]
I am using version 4.6.1 (latest for the time of writing) and making the following test

```
    public static void main(String[] args) {
        for (int i = 0; i < Integer.MAX_VALUE; i++) {
            UUID smaller = UuidCreator.getTimeOrderedWithMac();
            UUID bigger = UuidCreator.getTimeOrderedWithMac();
            Validate.isTrue(smaller.compareTo(bigger) < 0,
                    "\n%s\n%s", smaller, bigger);
        }
    }
```

When running that, I quickly get the assertion error. For example the 2 following IDs
```
1ecf7ad4-11c9-6d7f-9db5-02429e17b126
1ecf7ad4-11c7-6670-9db6-02429e17b126
```
Is it possible with that generator?

**Comment by fabiolimace** [comment]
Very interesting!

Fortunately, the clock sequence has been incremented from 00000000-0000-9db5-0000-000000000000 to 00000000-0000-9db6-0000-000000000000 to avoid duplicates.

On my PC it happened after 80 million in loop, simulating a huge batch.

I'll try to fix it after work.

**Comment by fabiolimace** [comment]
@tri-bao 

I found a solution to the problem.

I've also included a [new unit test](https://github.com/f4b6a3/uuid-creator/blob/72a5740eacfd6da5a5e651825c75f72a30a5c202/src/test/java/com/github/f4b6a3/uuid/factory/rfc4122/TimeOrderedFactoryTest.java#L197) based on your example.

The fix will be available soon in v5.0.0.

**Comment by fabiolimace** [comment]
Released v5.0.0.

Thanks for your report!
